security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -112,31 +112,30 @@ func newSDSService(st security.SecretManager, options *security.Options, pkpConf
 	// server in these cases.
 	go func() {
 		b := backoff.NewExponentialBackOff()
-		b.MaxElapsedTime = 0
-		for {
-			_, err := st.GenerateSecret(security.WorkloadKeyCertResourceName)
-			if err == nil {
-				break
-			}
-			sdsServiceLog.Warnf("failed to warm certificate: %v", err)
+		// context for both timeout and channel, whichever stops first, the context will be done
```

Comment:
I am not sure we ever want to timeout

Reply:
Forever retry is meaningless, this is actually pre generate to cache it. These logic can be refactor, maybe I can leave it for future